### ISSUE TITLE
Missing title on iframe in Chromium Browsers fails WCAG 2.2 2.4.1

### ISSUE THREAD

**Issue by accessabilly** [tinymce/tinymce#10620]
## 📝 Provide detailed reproduction steps (if any)
With TinyMCE 8.1 there is a new behaviour on the `<iframe>` Element containing the editor. There is a new config option `iframe_aria_text`. 

1. Install axe DevTools in your browser (Chrome, Edge)
2. Open [fiddle](https://fiddle.tiny.cloud/8R4ArFxpFa/5)
3. Open Dev Tools in Chrome > axe DevTools > Full Page Scan
4. Result should be: "41 Issues found". Ignore most other issues but "**Frames must have an accessible name**".
5. Second issue shows TinyMCE iframe that has no title attribute.

### ✔️ Expected result
The expected result is an **`<iframe>` with a `title`** attribute.
`<iframe id="mce_0_ifr" frameborder="0" allowtransparency="true" class="tox-edit-area__iframe" srcdoc="..." tile="Test title for Iframe">`

### ❌ Actual result
The actual result is an **`<iframe>` without a `title`** attribute.
`<iframe id="mce_0_ifr" frameborder="0" allowtransparency="true" class="tox-edit-area__iframe" srcdoc="...">`

### ❓ Possible solution
I know this is a wanted result as it is described on the [accessibility docs](https://www.tiny.cloud/docs/tinymce/latest/accessibility/#iframe_aria_text).
But this behaviour fails [WCAG 2.2 success criterion 2.4.1 "Bypass Blocks"](https://www.w3.org/TR/WCAG22/#bypass-blocks). The technique, how to make iframes accessible to all is described in [Technique H64](https://www.w3.org/WAI/WCAG22/Techniques/html/H64).

It's not enough to label the containing HTML documents `<body>`, because a blind user on the parent doc does not know where he/she is going or if this is info to skip, without any description of the `<iframe>`. I would like to ask you to rethink this behaviour, as it should have a `title` attribute in **all browsers**.
Thank you for your help and support!

## 📃 Other details
- Browser: Chromium Browsers like Chrome, Edge. Working correctly in Firefox though.
- OS: all
- First affected version: TinyMCE 8.1
- Worked in version: TinyMCE 8.0.2

**Comment by TheSpyder** [comment]
NVDA does not agree with the standard, it seems (ref #10490). I'll follow up to see if there are other approaches we can take.

**Comment by zuzanna-antas** [comment]
Our goal is to make TinyMCE as accessible as possible, ensuring that screen readers announce exactly what we intend them to. Screen reader behavior varies across browsers and operating systems. In this case, screen readers behaved in an unexpected way, so we adjusted our implementation (ref https://github.com/tinymce/tinymce/pull/10490) to achieve consistent and accurate announcements across platforms.

NVDA and JAWS — the screen readers officially supported by TinyMCE — announce both the iframe title and the aria-label applied to the body within the iframe in Chrome, and the iframe title in Firefox. However, macOS VoiceOver behaves differently: in Chrome and Safari, it correctly announces the aria-label applied to the body within the iframe, but not the iframe title itself. In Firefox, the opposite occurs — VoiceOver announces the title correctly but not the aria-label.

Previously, we included both a title and an aria-label, which caused duplicate announcements for NVDA and JAWS users. To avoid this and ensure the best experience for everyone, we now use the iframe title for Firefox and the aria-label for other browsers. This solution ensures that TinyMCE provides clear, accurate announcements for users on both macOS and Windows.

**Comment by accessabilly** [comment]
I appreciate your goal to make TinyMCE as accessible as possible very much! But I must say: it's a common misconception, that screen readers have to announce content in the same way and this should not be achieved via the use of ARIA or anything similar, because this can confuse users when their screen readers don't announce stuff in the expected way.

The people from Tetralogical just released [a great article on that (see Misconception 3)](https://tetralogical.com/blog/2025/10/14/common-misconceptions-about-screen-readers/#misconception-3%3A-all-screen-readers-work-in-the-same-way).

They also provide the great "Screen Reader HTML Support – Lookup", where you can find the different ways 5 screen readers treat HTML elements and [here, how the `<iframe>` element is announced](https://tetralogical.github.io/screen-reader-HTML-support/lookup/lookup.html#iframe). If TinyMCE would just conforms to this behaviour, it'll be great.

The usage of the `aria-label` here to force screen readers into a "preferred" behaviour could also break the "First Rule of ARIA use", that recommends using native HTML elements or attributes instead. Without having tested it: a `<title>` element inside the `<iframe>`'s document could have the same effect as an `aria-label` on the `<body>`. But this document has no `<title>`, and I can't see any reason why it should have not.

I personally think, the `title` attribute on an `<iframe>` is needed for all users, because they must have the right to decide, wether to enter the new document here or to skip it. Otherwise they would be forced to enter the document to find out, what it is about. And that's what [WCAG 2.2. 2.4.1](https://www.w3.org/WAI/WCAG22/Understanding/bypass-blocks) is about.

**Comment by zuzanna-antas** [comment]
Thank you for sharing the article! Note, that we're not trying to make all screen reader sound exactly the same. Our investigation showed that some screen readers don't announce title at all. We are making them announce something by setting an aria-label while avoiding double-speak on screen readers that announce both - title and aria-label. 

We think it's worth revisiting this so we will plan to investigate this issue further.

**Comment by kylane** [comment]
Unsure if this is helpful - but it fails the Deaque AxeTools tester:

<img width="1171" height="387" alt="Image" src="https://github.com/user-attachments/assets/92f103ff-e123-43bc-9f8b-321d25cb693f" />
